vulnerabilities/importers/openssl.py

```diff
@@ -20,101 +20,139 @@
-        if create_etag(data_src=self, url=self.url, etag_key="ETag"):
-            raw_data = self.fetch()
-            advisories = self.to_advisories(raw_data)
-            return self.batch_advisories(advisories)
-
-        return []
-
-    def fetch(self):
-        return requests.get(self.url).content
-
-    @staticmethod
-    def to_advisories(xml_response: str) -> Set[Advisory]:
-        advisories = []
-        pkg_name = "openssl"
-        pkg_type = "generic"
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        xml_response = requests.get(self.url).content
         root = ET.fromstring(xml_response)
-        for element in root:
-            if element.tag == "issue":
-                cve_id = ""
-                summary = ""
-                safe_pkg_versions = []
-                vuln_pkg_versions = []
-                ref_urls = []
-                for info in element:
-
-                    if info.tag == "cve":
-                        if info.attrib.get("name"):
-                            cve_id = "CVE-" + info.attrib.get("name")
-
-                        else:
-                            continue
-
-                    if cve_id == "CVE-2007-5502":
-                        # This CVE has weird version "fips-1.1.2".This is
-                        # probably a submodule. Skip this for now.
-                        continue
-
-                    if info.tag == "affects":
-                        # Vulnerable package versions
-                        vuln_pkg_versions.append(info.attrib.get("version"))
-
-                    if info.tag == "fixed":
-                        # Fixed package versions
-                        safe_pkg_versions.append(info.attrib.get("version"))
-
-                        if info:
-                            commit_hash = info[0].attrib["hash"]
-                            ref_urls.append(
-                                Reference(
-                                    url="https://github.com/openssl/openssl/commit/" + commit_hash
-                                )
-                            )
-                    if info.tag == "description":
-                        # Description
-                        summary = re.sub(r"\s+", " ", info.text).strip()
-
-                safe_purls = [
-                    PackageURL(name=pkg_name, type=pkg_type, version=version)
-                    for version in safe_pkg_versions
-                ]
-                vuln_purls = [
-                    PackageURL(name=pkg_name, type=pkg_type, version=version)
-                    for version in vuln_pkg_versions
-                ]
-
-                advisory = Advisory(
-                    vulnerability_id=cve_id,
-                    summary=summary,
-                    affected_packages=nearest_patched_package(vuln_purls, safe_purls),
-                    references=ref_urls,
+        for xml_issue in root:
+            if xml_issue.tag == "issue":
```

Comment:
Add log for the case, when it's not "issue"

Reply:
we can have something like this
```python
 logger.info(f'Looking for "issue" tag, found {xml_issue.tag!r} tag')
```